grammars/nearley.cson

```diff
@@ -90,18 +90,34 @@
     'patterns': [
       {
         'begin': '"'
-        'beginCaptures': '0': 'name': 'punctuation.definition.string.begin.php'
-        'contentName': 'meta.string-contents.quoted.double.php'
+        'beginCaptures': '0': 'name': 'punctuation.definition.string.begin.js'
         'end': '"'
-        'endCaptures': '0': 'name': 'punctuation.definition.string.end.php'
-        'name': 'string.quoted.double.php'
+        'endCaptures': '0': 'name': 'punctuation.definition.string.end.js'
+        'name': 'string.quoted.double.js'
+        'patterns': [
+          {
+            'include': '#string_escapes'
+          }
+          {
+            'match': '[^"]*$'
+            'name': 'invalid.illegal.string.js'
+          }
+        ]
+      }
+    ]
+
+  'string_escapes':
+    'patterns': [
+      {
+        'match': '\\\\u(?![A-Fa-f0-9]{4}|{[A-Fa-f0-9]+})[^\'"]*'
+        'name': 'invalid.illegal.unicode-escape.js'
+      }
+      {
+        'match': '\\\\u(?:[A-Fa-f0-9]{4})'
+        'name': 'constant.character.escape.js'
       }
       {
-        'begin': '\''
-        'beginCaptures': '0': 'name': 'punctuation.definition.string.begin.php'
-        'contentName': 'meta.string-contents.single.double.php'
-        'end': '\''
-        'endCaptures': '0': 'name': 'punctuation.definition.string.end.php'
-        'name': 'string.quoted.single.php'
+        'match': '\\\\["\'\\\\/bfnrt]'
+        'name': 'constant.character.escape.js'
```

Comment:
I think it should be made with `.nearley`, since we don't want js-specific styles applying to nearley comments and characters -- even though I guess that might be left for another PR :smiley: